From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 30 Aug 2022 22:55:00 +0200
Subject: [PATCH] Allow to disable colors in "runner" log format via env. vars

Using the same approach as the Logrus's default "text" formatter.

--- a/log/runner_formatter.go
+++ b/log/runner_formatter.go
@@ -3,6 +3,7 @@
 import (
 	"bytes"
 	"fmt"
+	"os"
 	"sort"
 
 	"github.com/sirupsen/logrus"
@@ -79,7 +80,7 @@
 		}
 	}
 
-	if f.DisableColors {
+	if !f.isColored() {
 		return "", "", prefix
 	}
 
@@ -98,6 +99,20 @@
 	}
 
 	return keys
+}
+
+// Based on isColored from https://github.com/sirupsen/logrus/blob/master/text_formatter.go.
+func (f *RunnerTextFormatter) isColored() bool {
+	if f.DisableColors {
+		return false
+	}
+
+	switch force, ok := os.LookupEnv("CLICOLOR_FORCE"); {
+	case ok && force == "0", os.Getenv("CLICOLOR") == "0":
+		return false
+	default:
+		return true
+	}
 }
 
 func SetRunnerFormatter() {
